### ISSUE TITLE
AdGuardHome keeps reporting errors and cause itself crash

### ISSUE THREAD

**Issue by lnc1995** [AdguardTeam/AdGuardHome#6937]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, AMD64 (aka x86_64)

### Installation

Docker

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

v0.107.48

### Action

I set the upstream DNS servers to "h3://8.8.8.8/dns-query" and "h3://1.1.1.1/dns-query" which is actually not reachable in China.
So the AdGuardHome depends on a proxy service to run，once the proxy offline,AdGuardHome will report erorrs crazy and then crash.
![image](https://github.com/AdguardTeam/AdGuardHome/assets/27806498/3a048d86-5533-4369-95ed-6781512c48ca)
Error info : [error] dnsproxy: https://8.8.8.8:443/dns-query: response received over udp: "requesting https://8.8.8.8:443/dns-query: Get_0rtt \"https://8.8.8.8:443/dns-query?dns=AAABBB\": timeout: no recent network activity"

### Expected result

Maybe the offical can add an option . After detecting the upstream server is unavailable, check again every customize mintues rather than checking so frequently.

### Actual result

 Keeping report erorrs and then crash.

**Comment by renatoyamane** [comment]
Are you using VPN?
I started to see these timeout erros when using OpenVPN (udp) on the router where the Adguard is also running

**Comment by lnc1995** [comment]
> Are you using VPN? I started to see these timeout erros when using OpenVPN (udp) on the router where the Adguard is also running

Yes,I'm using VPN,like i said , if the VPN down ,AdguardHome will report these errors.
